solutions/swb-app/src/projectRoutes.ts

```diff
@@ -141,4 +151,140 @@ export function setUpProjectRoutes(
       res.send(await projectService.updateProject(validatedRequest));
     })
   );
+
+  // add user to the project
+  router.post(
+    '/projects/:projectId/users/:userId',
+    wrapAsync(async (req: Request, res: Response) => {
+      const validatedRequest = validateAndParse<AssignUserToProjectRequest>(
+        AssignUserToProjectRequestParser,
+        {
+          projectId: req.params.projectId,
+          userId: req.params.userId,
+          role: req.body.role
+        }
+      );
+
+      const groupId = `${validatedRequest.projectId}#${req.body.role}`;
+
+      try {
+        const existingUser = await userService.getUser(validatedRequest.userId);
+
+        const isITAdmin = existingUser.roles.some((role) => role === 'ITAdmin');
+        if (isITAdmin) {
+          throw Boom.badRequest(
+            `IT Admin ${validatedRequest.userId} cannot be assigned to the project ${validatedRequest.projectId}`
+          );
+        }
+
+        // this call is needed to validate that project and group exists.
+        // If not - 404 will be returned
+        // TODO: use does group exist API
```

Comment:
i think haitao got the does group exist code into development so you can do this now if you want.

Reply:
Decided to leave as it is and removed comment; it's fire and forget create group call, otherwise we would need to check and create if group does not exist